x/validator-preference/msg_server_test.go

```diff
@@ -0,0 +1,111 @@
+package keeper_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	valPref "github.com/osmosis-labs/osmosis/v12/x/validator-preference"
+	"github.com/osmosis-labs/osmosis/v12/x/validator-preference/types"
+)
+
+func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
+	suite.SetupTest()
+
+	// setup 3 validators
+	valAddrs := suite.SetupMultipleValidators(3)
+
+	type param struct {
+		delegator   sdk.AccAddress
+		preferences []types.ValidatorPreference
+	}
```

Comment:
nit: consider unwrapping these from struct and using directly in the individual testing structs